### PR TITLE
callpackage a tool for the lazy

### PR DESCRIPTION
**Pull request by NobbZ**
- feat: create NoteBox
- post: callpackage a tool for the lazy

### CODE REVIEW

File: src/pages/posts/2022-08-24-callpackage.mdx [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+As you can see, `writeShellScriptBin` gets passed in by `callPackage` automatically.
+
+For this simple setup, having to create an extra file, seems to be a lot of
+boilerplate, though if you continue reading, you will see, it is worth it!
+
+## 1. Benefit: parametrized builds
+
+Now lets change the `default.nix`:
+
+```
```
Comment (NobbZ) [inline src/pages/posts/2022-08-24-callpackage.mdx:51]:
```suggestion
```nix
```

File: src/pages/posts/2022-08-24-callpackage.mdx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+Building and running the `folks` attribute, will give again a new version of the
+script.
+
+All the other parameters will remain the same as the have been when `hello` was
+instantiated.
+
+This is especially useful and often seen on packages that provide a whole lot of
+options to optimize the build.
+
+An example to mention here is the `neovim` attribute in nixpkgs, which has has
```
Comment (billewanick) [inline src/pages/posts/2022-08-24-callpackage.mdx:121]:
```suggestion
An example to mention here is the [`neovim` attribute in nixpkgs](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/neovim/utils.nix#L14-L41), which has has 
```

File: src/pages/posts/2022-08-24-callpackage.mdx [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+in nixpkgs already, it also gives you some things for free:
+
+1. parametrized builds
+2. overrideable builds
+3. cleaner implementation of large interdepending package sets
+
+## Further reading
+
+There is also `callPackages` and `lib.callPackages` which do a pretty similar
+thing, though they expect that the returnvalue is *not* a package, but a
```
Comment (billewanick) [inline src/pages/posts/2022-08-24-callpackage.mdx:192]: Is `returnvalue` a keyword in Nix? If not it should be split into `return value`

**Review by billewanick — COMMENTED** [review]
Great blog post, very insightful!
I've marked a bunch of spelling mistakes, and so other suggestions that I think would make it more readable

**Pull request merged**